queue/queue.go

```diff
@@ -35,11 +38,24 @@ type MetricsSource interface {
 }
 
 func New() (*Queue, error) {
-	return &Queue{}, nil
+	eventQueue := memqueue.NewQueue(logp.L(), memqueue.Settings{
+		Events: 1024,
+		// The event count and timeout for queue flushes is hard-coded to a placeholder
+		// for now, since that's what the existing Beats API understands. The plan is
+		// for these parameters to instead be controlled by the output as it reads
+		// the queue.
+		FlushMinEvents: 256,
+		FlushTimeout:   5 * time.Millisecond,
+	})
+	producer := eventQueue.Producer(beatsqueue.ProducerConfig{})
+	return &Queue{eventQueue: eventQueue, producer: producer}, nil
 }
 
 func (queue *Queue) Publish(event *api.Event) error {
-	return fmt.Errorf("couldn't publish: Queue.Publish is not implemented")
+	if !queue.producer.Publish(event) {
+		return fmt.Errorf("couldn't publish: queue is full")
```

Comment:
I think we'll want a dedicate error for this, since we'll probably want to check for it when this is used from the gRPC Publish call.

Reply:
I've moved it into a package variable -- is that enough, or are there additional steps to make sure gRPC understands it?